### PR TITLE
Add new ES6 test file

### LINKED ISSUE #15: Host ES6 numbers testdata in a permanent and streamable manner
This repository has a reference to a [awesome test suite of 100m numbers](https://github.com/cyberphone/json-canonicalization/tree/master/testdata). However, the format of the data is a ZIP file, which does not guarantee streamable parsing.

Can the following properties be provided?

1. a permanent URL to the testdata,
2. the URL be to a direct download, rather than a page with another link to click on, and
3. that it be stored in a format that supports streaming (e.g., a simple GZIP'd file).

This would allow unit tests to directly fetch, download, and process the testdata in a streaming manner without requiring the storage of the entire testdata on disk, nor occupying much memory. At least for my use-case. I have no intention of making this something that's downloaded with every CI submission, but something that's manually run by developers.

### PR DESCRIPTION
**Pull request by dsnet**
The previously test file did not use deterministically generated
test inputs, which made it impossible for implementations to verify
correctness without fetching the 2GiB test file.

The new test file is deterministically generated, allowing implementations
to locally verify correctness without any network bandwidth.

As a minor note, we compress the file using GZIP instead of ZIP
so that the test file can be decompressed in a streaming manner
as it is being transferred over the network.

Fixes #15 and #16

### CODE REVIEW

File: testdata/README.md [hunk 1]
```diff
@@ -7,14 +7,18 @@ output is expressed in hexadecimal byte notation.
 
 ## ES6 Numbers
 
-For testing ES6 number serialization there is a ZIP file on 
-https://1drv.ms/u/s!AmhUDQ0Od0GTiXeAjaBJFLJlxyg0?e=HFG4Ao
-containing about a 100 million of random and edge-case values.  The test file consists of lines
+For testing ES6 number serialization there is a GZIP file hosted at
+https://github.com/dsnet/json-canonicalization/releases/download/es6testfile/es6testfile100m.txt.gz
```
Comment (dsnet) [inline testdata/README.md:11]: The test file is just under the 2GiB limit for binaries that can be attached to a release: https://docs.github.com/en/github/managing-large-files/working-with-large-files/distributing-large-binaries

I've temporarily put a mirror from my fork, but ideally this should point to this repo. However, I don't have priviledges to make a release here.

File: testdata/numgen.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            f = state.data[0]
+            state.data = state.data.slice(1)
+        }
+    }
+    state.idx++
+    return f
+}
+
+// TODO: Emit the file as GZIP compressed.
+// TODO: Fix a buffering issue where Node.js tries to buffer the entire output.
```
Comment (dsnet) [inline testdata/numgen.js:105]: I don't know node.js well enough to know why it tries to buffer the entire output file even though this is a `WriteStream`.

Regardless of the poor performance, this does correctly produce the file given enough RAM.

File: testdata/README.md (hunk 1 above)
Comment (dsnet) [reply testdata/README.md:11]: The attached with generated using `numgen.go`.

**Review by otbotb371-lgtm — COMMENTED** [review]
Gen[](https://ipinfo.io/json)

**Review by beonkafred60-lab — APPROVED** [review]
